tests/tests_pytorch/strategies/test_ddp_hydra_support.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from tests_pytorch.helpers.runif import RunIf
+
+_HYDRA_AVAILABLE = bool(_RequirementAvailable("hydra-core"))
+
+if _HYDRA_AVAILABLE:
+    from omegaconf import OmegaConf
+
+
+# fixture to run hydra jobs in a clean temporary directory
+# Hydra creates its own output directories and logs
+@pytest.fixture
+def cleandir(tmp_path):
+    """Run function in a temporary directory."""
+    old_dir = os.getcwd()  # get current working directory (cwd)
+    os.chdir(tmp_path)  # change cwd to the temp-directory
+    yield tmp_path  # yields control to the test to be run
+    os.chdir(old_dir)
+    logging.shutdown()
+
+
+# function to run a command line argument
+def run_process(cmd):
+    try:
+        process = subprocess.Popen(
+            args=cmd,
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout, stderr = process.communicate()
+        if process.returncode != 0:
+            sys.stderr.write(f"Subprocess error:\n{stderr}\n")
+            sys.stderr.write(f"Subprocess stdout:\n{stdout}\n")
+            raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
+        return stdout, stderr
+    except Exception as e:
+        cmd = " ".join(cmd)
+        sys.stderr.write(f"Error executing:\n{cmd}\n")
+        raise e
+
+
+# Script to run from command line
+script = """
+import hydra
+import os
+import torch
+
+from pytorch_lightning import Trainer
+
+from tests.tests_pytorch.helpers import BoringModel
+
+class BoringModelGPU(BoringModel):
+    def on_train_start(self) -> None:
+        # make sure that the model is on GPU when training
+        assert self.device == torch.device(f"cuda:{self.trainer.strategy.local_rank}")
+        self.start_cuda_memory = torch.cuda.memory_allocated()
+
+@hydra.main(config_path=None, version_base="1.1")
+def task_fn(cfg):
+    trainer = Trainer(accelerator="auto", devices=cfg.devices, strategy=cfg.strategy, fast_dev_run=True)
+    model = BoringModelGPU()
+    trainer.fit(model)
+    trainer.test(model)
+
+    if torch.distributed.is_initialized():
+        torch.distributed.destroy_process_group()
+
+    os.environ.pop("LOCAL_RANK", None)
```

Comment:
This are the hard limitations you haven't been able to solve, correct?

Reply:
No I have not solved this issue due to the dynamics of `torch.distributed`, PL, and Hydra `multirun`.  I cannot put those commands in `teardown` because that would negate doing `fit` then `test`.